src/xhr.ts

```diff
@@ -41,6 +41,52 @@ export class JQueryRequestor extends Requestor {
   }
 }
 
+
+/**
+ * Uses fetch API to make Ajax requests
+ */
+export class FetchRequestor extends Requestor {
+  xhr<T>(settings: JQueryAjaxSettings): Promise<T> {
+    let url: URL = new URL(<string>settings.url);
+    let requestInit: RequestInit = {};
+    requestInit.method = settings.method;
+    requestInit.mode = 'cors';
+
+    if (settings.data) {
+      if (settings.method && settings.method.toUpperCase() === 'POST') {
+        requestInit.body = <string>settings.data;
+      } else {
+        let searchParams = new URLSearchParams(settings.data);
+        searchParams.forEach((value, key) => {
+          url.searchParams.append(key, value);
+        });
+      }
+    }
+
+    // Calculate 'Accept' header value
+    let headers = settings.headers || {};
+    if (settings.dataType && settings.dataType.toLowerCase() === 'json') {
+      headers['Accept'] = 'application/json, text/javascript, */*; q=0.01';
+    } else {
+      headers['Accept'] = '*/*';
+    }
+
+    // Set the request headers
+    requestInit.headers = {};
+    for (let i in headers) {
+      requestInit.headers[i] = <string>headers[i];
+    }
+
+    return fetch(url.toString(), requestInit).then(response => {
+      if (settings.dataType && settings.dataType.toLowerCase() === 'json') {
```

Comment:
There is no check to see if the request was successful. Something like `response.ok` and/or `response.status >= 200 && response.status < 300` should probably be done. Throw the response if it failed?

From [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch):

> The Promise returned from fetch() won’t reject on HTTP error status even if the response is an HTTP 404 or 500. Instead, it will resolve normally (with ok status set to false), and it will only reject on network failure or if anything prevented the request from completing.

Reply:
Thanks for  catching that @someone1. You should wrap it with a `Promise` and throw an `AppAuthError` wrapping the underlying `Response`.

Reply:
Actually, after re-reading it, I think that the default mode is `cors` so setting it explicitly shouldn't hurt but I don't think its required. I know I don't use this setting in any of my code that interacts with APIs and have no issues.

Reply:
Thanks @someone1 , good catch!